### PR TITLE
Fix Issue #548

### PR DESCRIPTION
**Pull request by karandesai-96**
This PR is a fix of issue #548 as well as a simplification of PR #540. Previous implementation caused ocassional segfaults in subsequent test runs. Here, the complete file **enum.py** has been removed and corresponding variables from C enums have been defined as macros in **struct.py**. This ensures that the code is still readable as before.

### CODE REVIEW

File: tardis/montecarlo/struct.py [hunk 1]
```diff
@@ -22,7 +21,7 @@ class RPacket(Structure):
         ('d_boundary', c_double),
         ('d_cont', c_double),
         ('next_shell_id', c_int64),
-        ('status', RPacketStatus),
+        ('status', c_int),
```
Comment (yeganer) [inline tardis/montecarlo/struct.py]: Use `c_status_t` here and define it as `c_status_t = c_int` above.

File: tardis/montecarlo/struct.py [hunk 2]
```diff
@@ -85,7 +84,7 @@ class StorageModel(Structure):
         ('t_electrons', POINTER(c_double)),
         ('l_pop', POINTER(c_double)),
         ('l_pop_r', POINTER(c_double)),
-        ('cont_status', ContinuumProcessesStatus),
+        ('cont_status', c_int),
```
Comment (yeganer) [inline tardis/montecarlo/struct.py]: `c_cont_status_t`

File: tardis/montecarlo/struct.py (hunk 1 above)
Comment (karandesai-96) [inline tardis/montecarlo/struct.py]: - [X] Done. ( `c_rpacket_status_t` )

File: tardis/montecarlo/struct.py (hunk 2 above)
Comment (karandesai-96) [inline tardis/montecarlo/struct.py]: - [X] Done.

**Comment by karandesai-96** [comment]
Great, everything is neat.
@yeganer I believe this resolves #548 and silences down segfaults occuring in #544.

File: tardis/montecarlo/struct.py [hunk 3]
```diff
@@ -1,5 +1,8 @@
 from ctypes import Structure, POINTER, c_int, c_int64, c_double, c_ulong
 
+c_rpacket_status_t = c_int
+c_cont_status_t = c_int
+
```
Comment (yeganer) [inline tardis/montecarlo/struct.py]: you missed `c_error_t` here

Comment (karandesai-96) [inline tardis/montecarlo/struct.py]: It isn't used in structs right now, though declaration would be helpful.

Comment (karandesai-96) [inline tardis/montecarlo/struct.py]: - [X] Done. ( `c_tardis_error_t` )

**Comment by yeganer** [comment]
@tardis-sn/tardis-core Can someone please sign this off? This resolves issue #548.

**Pull request merged**